python/cudf/cudf/_lib/cpp/groupby.pxd

```diff
@@ -5,13 +5,18 @@ from libcpp.memory cimport unique_ptr
 from libcpp.pair cimport pair
 from libcpp cimport bool
 
+from cudf._lib.cpp.libcpp.functional cimport reference_wrapper
 from cudf._lib.cpp.table.table cimport table
 from cudf._lib.cpp.table.table_view cimport table_view
 from cudf._lib.cpp.column.column_view cimport column_view
 from cudf._lib.cpp.column.column cimport column
 from cudf._lib.cpp.aggregation cimport aggregation
+from cudf._lib.cpp.scalar.scalar cimport scalar
 from cudf._lib.cpp.types cimport size_type, order, null_order, null_policy
 
+# workaround for https://github.com/cython/cython/issues/3885
+ctypedef const scalar constscalar
```

Comment:
This should be fixed as of Cython 0.29.22

Reply:
Will try to remove.

Reply:
So previously @shwina 's PR managed to fix `const` usage inside `cdef` statements. Here we need to use it inline as an expression (at line 224)
```cython
 c_fill_values.push_back(
    reference_wrapper[constscalar](d_slr.get_raw_ptr()[0])
)
```
If we remove the `ctypedef` the parser still complains:
```
Error compiling Cython file:
------------------------------------------------------------
...
        c_fill_values.reserve(num_col)
        for val, col in zip(fill_values, values._columns):
            d_slr = as_device_scalar(val, dtype=col.dtype)
            d_slrs.append(d_slr)
            c_fill_values.push_back(
                reference_wrapper[const scalar](d_slr.get_raw_ptr()[0])
                                       ^
------------------------------------------------------------

cudf/_lib/groupby.pyx:224:40: Expected ']', found 'scalar'
```
My cython version:
```
cython                    0.29.23          py37hcd2ae1e_0    conda-forge
```
Reported issue: https://github.com/cython/cython/issues/4180